packages/contracts-bedrock/src/L1/OPContractsManagerStandardValidator.sol

```diff
@@ -798,4 +826,51 @@ contract OPContractsManagerStandardValidator is ISemver {
 
         return finalErrors;
     }
+
+    /// @notice Internal function to read all information from a dispute game while supporting both v1 and v2 dispute
+    /// games.
+    function _decodeDisputeGameImpl(
+        IPermissionedDisputeGame _game,
+        bytes memory _gameArgsBytes,
+        bool _isPermissioned
+    )
+        internal
+        view
+        returns (DisputeGameImplementation memory gameImpl_)
+    {
+        bytes32 absolutePrestate;
+        address vm;
+        address asr;
+        address weth;
+        uint256 l2ChainId;
+        address challengerAddress;
+        if (DevFeatures.isDevFeatureEnabled(devFeatureBitmap, DevFeatures.DEPLOY_V2_DISPUTE_GAMES)) {
+            (absolutePrestate, vm, asr, weth, l2ChainId,, challengerAddress) = LibGameArgs.decode(_gameArgsBytes);
```

Comment:
It seems the intention is for `StandardValidator` to avoid reverting even if there are validation failures.  So, I think we should validate the bytelength of gameArgs before decoding and if the length is invalid, we should set an error string for this. 

Fine to just add a `TODO(#17267)` here though - I can follow-up as part of [this ticket](https://github.com/ethereum-optimism/optimism/issues/17267).

Reply:
I'm not sure that's the intent of `StandardValidator`. The stdvalidator only avoids reverting for explicit checks. Like when a proxyAdmin isn't what it expects.
Interacting with contracts that have unexpectedly missing ABIs or calls that revert internally in contracts aren't try-caught.  And we want those calls to revert as they suggest there's a bug in the std validator itself.

Reply:
std validator now checks the game args length when appropriate.